libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -468,61 +470,36 @@ int HTTPClient::writeToStream(Stream * stream) {
 
     // get length of document (is -1 when Server sends no Content-Length header)
     int len = _size;
-    int bytesWritten = 0;
-
-    size_t buff_size = HTTP_TCP_BUFFER_SIZE;
-
-    // if possible create smaller buffer then HTTP_TCP_BUFFER_SIZE
-    if((len > 0) && (len < HTTP_TCP_BUFFER_SIZE)) {
-        buff_size = len;
-    }
-
-    // create buffer for read
-    uint8_t * buff = (uint8_t *) malloc(buff_size);
-
-    if(buff) {
-        // read all data from server
-        while(connected() && (len > 0 || len == -1)) {
-
-            // get available data size
-            size_t size = _tcp->available();
-
-            if(size) {
-                int c = _tcp->readBytes(buff, ((size > buff_size) ? buff_size : size));
-
-                // write it to Stream
-                int w = stream->write(buff, c);
-                bytesWritten += w;
-                if(w != c) {
-                    DEBUG_HTTPCLIENT("[HTTP-Client][writeToStream] short write asked for %d but got %d\n", c, w);
-                    break;
-                }
-
-                if(len > 0) {
-                    len -= c;
+    int ret = 0;
+
+    if(_transferEncoding == HTTPC_TE_IDENTITY) {
+        ret = writeToStreamDataBlock(stream, len);
+    } else if(_transferEncoding == HTTPC_TE_CHUNKED) {
+        while(1) {
+            String chunkHeader = _tcp->readStringUntil('\n');
+            chunkHeader.trim(); // remove \r
+
+            // read size of chunk
+            len = (uint32_t) strtol((const char *) chunkHeader.c_str(), NULL, 16);
+            DEBUG_HTTPCLIENT("[HTTP-Client] read chunk len: %d\n", len);
+            if(len > 0) {
+                int r = writeToStreamDataBlock(stream, len);
```

Comment:
Should this check if r != len to handle short writes?  If there's a short write then it looks like this'll continue to iterate over the response.